vaadin-map-flow-parent/vaadin-map-flow/src/main/resources/META-INF/resources/frontend/vaadin-map/synchronization/index.js

```diff
@@ -21,6 +24,16 @@ import {
 import { synchronizeIcon, synchronizeFill, synchronizeStroke, synchronizeStyle } from './styles.js';
 import { convertToCoordinateArray, synchronizeCollection } from './util.js';
 
+/**
+ * Fallback text style to use for features that don't have a custom one
+ */
+const fallbackTextStyle = new Text({
+  font: '13px sans-serif',
+  offsetY: 10,
+  fill: new Fill({ color: '#333' }),
+  stroke: new Stroke({ color: '#fff', width: 3 })
+});
```

Comment:
I got a bit stuck here trying to understand one thing about this shared `Text` object and would like to share my understanding to check whether I got it right.

For all `Feature` objects that don't have a custom text style, we are setting `fallbackTextStyle`. So, all objects will share the same instance, which means that when we put the actual text on the line `#99`, we are changing the same object. From what I could test, it works with multiple labels because:
1. When a `Feature` have its label changed, OL will call `target.getStyle()` for all the `Feature` objects
2. Immediately after getting the `Style` object, OL will proceed to render the feature and its label and, at some point, it copies the text string to another object it creates.

So, although we at the end are changing the text on the same object, that doesn't seem to be an issue here.

----

If I got everything correctly, I would propose to improve the IT a little bit by adding more than one `Feature` and making changes on theirs label in different orders, just to make sure that no undesired behavior happens.

Reply:
Yep, that is correct. It's a common pattern in OpenLayers, for example in this demo: https://openlayers.org/en/latest/examples/street-labels.html. My assumption is that for each feature, OL calls the style function, uses the returned style to immediately render that feature on the canvas, and then discard it.

Adding an IT for multiple labels make senses. Changing labels in different orders probably doesn't make a difference due to how the sync works, which always collects children in the same order, so I would skip that.

Reply:
Changed the IT to cover multiple features / labels.